knowledge-graph/src/main/scala/ch/datascience/knowledgegraph/datasets/rest/DatasetsFinder.scala

```diff
@@ -118,15 +118,27 @@ private class IODatasetsFinder(
         |            ?luceneDsId schema:creator ?id;
         |                        rdf:type <http://schema.org/Dataset>;
         |                        renku:topmostSameAs ?sameAs.
+        |                        
         |          }
         |        }
         |      } {
         |        ?dsId rdf:type <http://schema.org/Dataset>;
         |              renku:topmostSameAs ?sameAs;
-        |              schema:isPartOf ?projectId.
+        |              schema:isPartOf ?projectId ;
+        |              prov:atLocation ?location .
+        |              
+        |        BIND(CONCAT(?location, "/metadata.yml") AS ?metaDataLocation).
         |        FILTER NOT EXISTS {
-        |          ?someId prov:wasDerivedFrom/schema:url ?dsId.
-        |          ?someId schema:isPartOf ?projectId.
+        |            # Removing dataset that have an activity that invalidates them
+        |            ?deprecationEntity rdf:type <http://www.w3.org/ns/prov#Entity>;
+        |                               prov:atLocation ?metaDataLocation ;
+        |                               prov:wasInvalidatedBy ?invalidationActivity ;
+        |                               schema:isPartOf ?projectId .
+        |        }
+        |        
+        |        FILTER NOT EXISTS {
```

Comment:
Did you determine that these two filter clauses are executed together or sequentially? Does it matter?

Reply:
It could matter for performance, they should be executed sequentially. This is some kind of an OR statement so the order does not really matter. We want to filter out the dataset with an invalidation activity or the dataset that are derivedFrom.